components/gitpod-db/src/typeorm/webhook-event-db-impl.ts

```diff
@@ -52,19 +52,17 @@ export class WebhookEventDBImpl implements WebhookEventDB {
         return query.getMany();
     }
 
-    public async deleteOldEvents(ageInDays: number, limit?: number): Promise<void> {
+    public async deleteOldEvents(ageInDays: number, limit: number): Promise<void> {
         const repo = await this.getRepo();
         const d = new Date();
         d.setDate(d.getDate() - ageInDays);
         const expirationDate = d.toISOString();
-        const query = repo
+        await repo
             .createQueryBuilder()
-            .update()
-            .set({ deleted: true })
-            .where("creationTime <= :expirationDate", { expirationDate });
-        if (typeof limit === "number") {
-            query.limit(limit);
-        }
-        await query.execute();
+            .limit(limit)
+            .where("creationTime <= :expirationDate", { expirationDate })
+            .delete();
```

Comment:
This was previously reverted in https://github.com/gitpod-io/gitpod/commit/e0f8cd3251dae5003d71fe2ca898387900da3d92

The problem was the extra `execute` statement, which is now removed.

Reply:
IIRC we also found out that limit is ignored in delete statements, no?

Reply:
alternative could be to delete with a limited sub select:

```
DELETE t.* FROM table_name t 
JOIN (
  SELECT id FROM table_name 
  ORDER BY id 
  LIMIT 10
) t1 ON t.id = t1.id;
```

Reply:
DELETE ... LIMIT should just work, at least in MySQL world. Not sure if typeorm breaks stuff, though...

Reply:
Oh sorry for spreading FUD here. It should indeed work for such simple delete statements. It just doesn't work in DELETES with joins 
> You cannot use ORDER BY or LIMIT in a multiple-table DELETE.
(see https://dev.mysql.com/doc/refman/5.7/en/delete.html)

Reply:
I'm fighting dbtests in my workspace a bit, trying to validate the exact query it uses for this. Will get back here once I have that.

Reply:
Looks like it indeed ignores the limit. It executes 
```
DELETE FROM `d_b_webhook_event` WHERE `creationTime` <= :expirationDate
```

Reply:
I've rewritten the query